src/StaticATokenLM.sol

```diff
@@ -340,28 +337,20 @@ contract StaticATokenLM is
     return _convertToAssets(shares, Rounding.DOWN);
   }
 
-  ///@inheritdoc IERC4626
-  function maxDeposit(address) public view virtual returns (uint256) {
-    return type(uint256).max;
-  }
-
   ///@inheritdoc IERC4626
   function maxMint(address) public view virtual returns (uint256) {
-    return type(uint256).max;
+    uint256 assets = maxDeposit(address(0));
+    return assets == 0 ? 0 : _convertToShares(assets, Rounding.DOWN);
```

Comment:
I would either remove this ternary operator completely or move it inside _convertToShares. But better to remove imo